src/types/boxed.rs

```diff
@@ -0,0 +1,343 @@
+///     let person = RefCell::new(Person::new(name));
+/// 
+///     Ok(cx.boxed(person))
+/// }
+/// 
+/// fn person_set_name(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+///     let person = cx.argument::<BoxedPerson>(0)?;
+///     let mut person = person.borrow_mut();
+///     let name = cx.argument::<JsString>(1)?.value(&mut cx);
+/// 
+///     person.set_name(name);
+/// 
+///     Ok(cx.undefined())
+/// }
+/// 
+/// fn person_greet(mut cx: FunctionContext) -> JsResult<JsString> {
+///     let person = cx.argument::<BoxedPerson>(0)?;
+///     let person = person.borrow();
+///     let greeting = person.greet();
+/// 
+///     Ok(cx.string(greeting))
+/// }
+pub struct JsBox<T: Send + 'static> {
+    local: raw::Local,
+    // `JsBox` cannot verify the lifetime. Store a raw pointer to force uses
+    // to be marked unsafe. In practice, it can be treated as `'static` but
+    // should only be exposed as part of a `Handle` tied to a `Context` lifetime.
+    internal: *const T,
+}
+
+// Custom `Clone` implementation since `T` might not be `Clone`
+impl<T: Send + 'static> Clone for JsBox<T> {
+    fn clone(&self) -> Self {
+        JsBox {
+            local: self.local,
+            internal: self.internal,
+        }
+    }
+}
+
+impl<T: Send + 'static> Copy for JsBox<T> {}
+
+impl<T: Send + 'static> Value for JsBox<T> { }
+
+impl<T: Send + 'static> Managed for JsBox<T> {
+    fn to_raw(self) -> raw::Local {
+        self.local
+    }
+
+    fn from_raw(env: Env, h: raw::Local) -> Self {
+        let v = unsafe {
+            external::deref::<BoxAny>(env.to_raw(), h)
+                .map(|v| &*v)
+        };
+
+        let internal = v
+            .and_then(|v| v.downcast_ref())
+            .expect("Expected type to already be validated");
+
+        Self {
+            local: h,
+            internal,
+        }        
+    }
+}
+
+impl<T: Send + 'static> ValueInternal for JsBox<T> {
+    fn name() -> String {
+        any::type_name::<Self>().to_string()
+    }
+
+    fn is_typeof<Other: Value>(env: Env, other: Other) -> bool {
+        let v = unsafe {
+            external::deref::<BoxAny>(env.to_raw(), other.to_raw())
+                .map(|v| &*v)
+        };
+
+        v.map(|v| v.is::<T>()).unwrap_or(false)
+    }
+
+    fn downcast<Other: Value>(env: Env, other: Other) -> Option<Self> {
+        let local = other.to_raw();
+        let v = unsafe {
+            external::deref::<BoxAny>(env.to_raw(), local)
+                .map(|v| &*v)
+        };
+
+        v.and_then(|v| v.downcast_ref())
+            .map(|internal| Self {
+                local,
+                internal,
+            })
+    }
+}
+
+impl<T: Finalize + Send + 'static> JsBox<T> {
+    /// Constructs a new `JsBox` containing `value`.
+    pub fn new<'a, C>(cx: &mut C, value: T) -> Handle<'a, JsBox<T>>
+    where
+        C: Context<'a>,
+        T: Send + 'static,
+    {
+        fn finalizer<U: Finalize + 'static>(env: raw::Env, data: BoxAny) {
```

Comment:
Can we document in a comment what this function is for? I think the idea is that it runs the finalization of the Rust data whenever the GC decides to collect the `JsBox` wrapper object, yes?